### ISSUE TITLE
Clarification about README note re. using slz on small data

### ISSUE THREAD

**Issue by jberryman** [wtarreau/libslz#3]
From the README:

> It doesn't hurt compression performance when it is fed in large enough chunks (at
least a few kB at once), but it must not be used if it has to read less than
a few kB at a time, or the caller will have to implement the input buffer to
maximize the compression ratio. For example, an HTTP gateway may defragment all
the parts from a chunked-encoded response buffer into a linear buffer that is
then fed at once to SLZ. This temporary buffer would then be static and shared
between all streams running on the same thread.

Do I understand correctly that “must not” here means “Or else compression will be poor”? Or will something worse happen?

Also would it be possible to give an example of how you’d use the API in the way described here? I'm struggling a bit to follow

Thanks for your work on this library!

**Comment by wtarreau** [comment]
You're absolutely right, the wording is really poor. I should have written "it should preferably not be used ...". As you guessed, it's only that the compression ratio will be quite poor without buffering.

**Comment by wtarreau** [comment]
Finally fixed, so I can close :-)
